examples/consumer.cpp

```diff
@@ -56,15 +73,63 @@ class Consumer
 int
 main(int argc, char* argv[])
 {
-  if (argc != 2) {
-    std::cout << "usage: " << argv[0] << " <service-name> "
-              << std::endl;
-    return 1;
-  }
+  std::string serviceName;
+  int contFlag = -1;
+
+  namespace po = boost::program_options;
+  po::options_description visibleOptDesc("Options");
+
+  visibleOptDesc.add_options()
+    ("help,h",      "print this message and exit")
+    ("serviceName,s", po::value<std::string>(&serviceName)->required(), "Service name to fetch service info")
+    ("continuous,c", po::value<int>(&contFlag), "continuous discovery, 1 for true 0 for false")
+  ;
+
+  try
+  {
+    po::variables_map optVm;
+    po::store(po::parse_command_line(argc, argv, visibleOptDesc), optVm);
+    po::notify(optVm);
+
+    if (optVm.count("continuous")) {
+        if (contFlag != ndnsd::discovery::OPTIONAL and contFlag != ndnsd::discovery::REQUIRED) 
+        {
+          std::cout << "'c' must be either '0' or '1', default i.e. '0' will be used" << std::endl;
+        }
+      }
+    else
+      contFlag = 0;
+
+    if (optVm.count("serviceName")) {
+      if (serviceName.empty())
+      {
+        std::cerr << "ERROR: serviceName cannot be empty" << std::endl;
+        usage(visibleOptDesc);
+      }
+    }
 
+  }
+  catch (const po::error& e) {
+    std::cerr << "ERROR: " << e.what() << std::endl;
+    usage(visibleOptDesc);
+  }
+  // TODO: protocol shouldn't be hard-coded.
   std::map<char, uint8_t> flags;
```

Comment:
What I meant was to replay this std::map with an options class. Where are the new changes?